scripts/build_renewable_profiles.py

```diff
@@ -252,13 +252,18 @@
 
     # filter plants for hydro
     if snakemake.wildcards.technology.startswith("hydro"):
+        country_shapes = gpd.read_file(paths.country_shapes)
         hydrobasins = gpd.read_file(resource["hydrobasins"])
-        merged_hydrobasin_shape = unary_union(hydrobasins.geometry)
+        hydrobasins = hydrobasins[[
```

Comment:
Can you linger on the reason for this change? What was the previous issue?

Reply:
The unary_union for the world hydrobasins was taking forever to compute.
Therefore, I preferred to decompose the problem into:
1. remove from hydrobasins dataframe the shapes that do not intersect to the countries of interest
2. perform the computations for only the buses that lay in a hydrobasin shape